packages/core/src/version-utils.ts

```diff
@@ -0,0 +1,106 @@
+import { normalizeErrorString } from './outcomes';
+
+export const GITHUB_RELEASES_URL = 'https://api.github.com/repos/medplum/medplum/releases';
+
+export type ReleaseManifest = { tag_name: string; assets: { name: string; browser_download_url: string }[] };
+
+const releaseManifests = new Map<string, ReleaseManifest>();
+
+/**
+ * Clears the locally-cached `ReleaseManifest`s for all versions.
+ */
+export function clearReleaseCache(): void {
+  releaseManifests.clear();
+}
+
+/**
+ * Asserts that a given candidate is a `ReleaseManifest`.
+ * @param candidate - An object assumed to be a `ReleaseManifest`.
+ */
+export function assertReleaseManifest(candidate: unknown): asserts candidate is ReleaseManifest {
+  const manifest = candidate as ReleaseManifest;
+  if (!manifest.tag_name) {
+    throw new Error('Manifest missing tag_name');
+  }
+  const assets = manifest.assets;
+  if (!assets?.length) {
+    throw new Error('Manifest missing assets list');
+  }
+  for (const asset of assets) {
+    if (!asset.browser_download_url) {
+      throw new Error('Asset missing browser download URL');
+    }
+    if (!asset.name) {
+      throw new Error('Asset missing name');
+    }
+  }
+}
+
+/**
+ * @param version - The version to fetch. If no `version` is provided, defaults to the `latest` version.
+ * @returns - The manifest for the specified or latest version.
+ */
+export async function fetchVersionManifest(version?: string): Promise<ReleaseManifest> {
+  let manifest = releaseManifests.get(version ?? 'latest');
+  if (!manifest) {
+    const versionTag = version ? `tags/v${version}` : 'latest';
+    const res = await fetch(`${GITHUB_RELEASES_URL}/${versionTag}`);
+    if (res.status !== 200) {
+      let message: string | undefined;
+      try {
+        message = ((await res.json()) as { message: string }).message;
+      } catch (err) {
+        console.error(`Failed to parse message from body: ${normalizeErrorString(err)}`);
+      }
+      throw new Error(
+        `Received status code ${res.status} while fetching manifest for version '${version ?? 'latest'}'. Message: ${message}`
+      );
+    }
+    const response = (await res.json()) as ReleaseManifest;
+    assertReleaseManifest(response);
+    manifest = response;
+    releaseManifests.set(version ?? 'latest', manifest);
+    if (!version) {
+      releaseManifests.set(manifest.tag_name.slice(1), manifest);
+    }
+  }
+  return manifest;
+}
+
+/**
+ * Tests that a given version string follows the basic semver pattern of `<int>.<int>.<int>`, which is used for Medplum versions.
+ *
+ * @param version - A version string that should be tested for valid semver semantics.
+ * @returns `true` if `version` is a valid semver version that conforms to the Medplum versioning system, otherwise `false`.
+ */
+export function isValidMedplumSemver(version: string): boolean {
```

Comment:
Why `isValidMedplumSemver` vs `isValidSemver`?

Reply:
SemVer technically includes allowances of a bunch of extra symbols:
`^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$`
https://semver.org/#:~:text=Additional%20labels%20for%20pre%2Drelease%20and%20build%20metadata%20are%20available%20as%20extensions%20to%20the%20MAJOR.MINOR.PATCH%20format.

This is just a check for the simple `MAJOR.MINOR.PATCH` we use

Reply:
Since we are exporting it from `@medplum/core` I don't want users to believe something is NOT semver because our function returns false.

Alternatively we could provide the full regex implementation provided by the `SemVer` site, I'm ok with doing that